packages/block-library/src/social-links/edit.js

```diff
@@ -58,23 +59,33 @@ export function SocialLinksEdit( props ) {
 
 	const {
 		iconBackgroundColorValue,
+		customIconBackgroundColor,
 		iconColorValue,
 		openInNewTab,
 		size,
 		layout,
 	} = attributes;
+
 	const usedLayout = layout || getDefaultBlockLayout( name );
 
 	// Remove icon background color if logos only style selected.
-	const logosOnly =
-		attributes.className?.indexOf( 'is-style-logos-only' ) >= 0;
+	const logosOnly = attributes.className
+		? attributes.className.indexOf( 'is-style-logos-only' ) >= 0
+		: false;
 	useEffect( () => {
 		if ( logosOnly ) {
+			setbackgroundBackup( {
+				iconBackgroundColor,
+				iconBackgroundColorValue,
+				customIconBackgroundColor,
+			} );
```

Comment:
```suggestion
			backgroundBackup.current = {
				iconBackgroundColor,
				iconBackgroundColorValue,
				customIconBackgroundColor,
			};
```
To continue from the comment about switching to a ref.